// @transak/transak-sdk/lib/index.js
import w from"events";var s=(i=>(i.TRANSAK_WIDGET_INITIALISED="TRANSAK_WIDGET_INITIALISED",i.TRANSAK_ORDER_CREATED="TRANSAK_ORDER_CREATED",i.TRANSAK_ORDER_SUCCESSFUL="TRANSAK_ORDER_SUCCESSFUL",i.TRANSAK_ORDER_CANCELLED="TRANSAK_ORDER_CANCELLED",i.TRANSAK_ORDER_FAILED="TRANSAK_ORDER_FAILED",i.TRANSAK_WALLET_REDIRECTION="TRANSAK_WALLET_REDIRECTION",i.TRANSAK_WIDGET_CLOSE_REQUEST="TRANSAK_WIDGET_CLOSE_REQUEST",i.TRANSAK_WIDGET_CLOSE="TRANSAK_WIDGET_CLOSE",i))(s||{});import y from"query-string";var f={name:"@transak/transak-sdk",version:"4.0.2",description:"Transak SDK that allows you to easily integrate fiat on/off ramp",type:"module",types:"lib/index.d.ts",main:"lib/index.cjs",exports:{".":{import:{types:"./lib/index.d.ts",default:"./lib/index.js"},require:{types:"./lib/index.d.cts",default:"./lib/index.cjs"}}},engines:{node:">=20.0.0"},packageManager:"pnpm@10.16.1+sha512.0e155aa2629db8672b49e8475da6226aa4bdea85fdcdfdc15350874946d4f3c91faaf64cbdc4a5d1ab8002f473d5c3fcedcd197989cf0390f9badd3c04678706",files:["lib/**/*"],scripts:{eslint:"eslint . --ext .ts --fix",build:"tsc && tsup",prepack:"pnpm build",packDev:"pnpm pack"},author:"Transak",license:"ISC",homepage:"https://docs.transak.com/docs/web-integration#transak-sdk-reactvueangularts",dependencies:{events:"^3.3.0","query-string":"^9.3.0"},devDependencies:{"@types/events":"^3.0.3","@typescript-eslint/eslint-plugin":"^7.18.0","@typescript-eslint/parser":"^7.18.0",eslint:"~8.57.1","eslint-config-airbnb-base":"^15.0.0","eslint-config-airbnb-typescript":"^18.0.0","eslint-config-prettier":"^10.1.8","eslint-plugin-eslint-comments":"^3.2.0","eslint-plugin-import":"^2.32.0","eslint-plugin-no-relative-import-paths":"^1.6.1","eslint-plugin-prettier":"^5.5.4","eslint-plugin-promise":"^7.2.1",prettier:"^3.6.2",tsup:"^8.5.0",typescript:"~5.5.4"},keywords:["crypto","cryptocurrency","fiat","on","off","ramp","sdk","ts","js"]};var l=t=>{if(!t)return!1;try{return!!new URL(t)}catch{return!1}},g=t=>{if(!l(t))throw new Error("Invalid URL");return t};function d(t){let{name:e,version:n}=f,{referrer:r,widgetUrl:o}=t||{};if(!l(r)||!l(o))throw new Error("referrer and widgetUrl are required and must be valid URLs");let a=y.stringifyUrl({url:o,query:{sdkName:e,sdkVersion:n}},{arrayFormat:"comma"});return g(a)}function c(t){let e=document.createElement("iframe");return Object.assign(e,{id:"transakIframe",allow:"camera;microphone;payment",src:t}),e}function h(){return`
    #transakIframe{
      width: 100%;
      height: 100%;
      border: none;
    }
  `}function E(){let t=document.createElement("style");return t.innerHTML=h(),document.getElementsByTagName("head")[0]?.appendChild(t),t}function u(t){let e=E(),n=c(d(t));if(t.containerId){let r=document.getElementById(t.containerId);if(r)r.appendChild(n);else throw new Error("[Transak SDK] => Please enter a valid containerId")}return{styleElement:e,iframeElement:n}}var T=`
  <svg id="transakCloseIcon" viewBox="0 0 612 612" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
    <path d="M306,0C136.992,0,0,136.992,0,306s136.992,306,306,306c168.988,0,306-137.012,306-306S475.008,0,306,0z M414.19,387.147
    c7.478,7.478,7.478,19.584,0,27.043c-7.479,7.478-19.584,7.478-27.043,0l-81.032-81.033l-81.588,81.588
    c-7.535,7.516-19.737,7.516-27.253,0c-7.535-7.535-7.535-19.737,0-27.254l81.587-81.587l-81.033-81.033
    c-7.478-7.478-7.478-19.584,0-27.042c7.478-7.478,19.584-7.478,27.042,0l81.033,81.033l82.181-82.18
    c7.535-7.535,19.736-7.535,27.253,0c7.535,7.535,7.535,19.737,0,27.253l-82.181,82.181L414.19,387.147z" />
  </svg>
`;function I(t,e,n){return`
    #transakRoot {
      z-index: 9997;
      position: fixed;
      top: 0;
      right: 0;
      bottom: 0;
      left: 0;
      background: rgba(0, 0, 0, 0.6);
    }

    #transakModal {
      z-index: 9998;
      position: fixed;
      width: ${e};
      height: calc(${n} - 24px);
      top: 50%;
      left: 50%;
      transform: translate(-50%, calc(-50% - 12px));
      margin-top: 24px;
    }

    #transakCloseIcon {
      z-index: 9999;
      position: absolute;
      width: 36px;
      height: 36px;
      top: -24px;
      right: 0;
      transition: 0.5s;
      color: #${t};
      background: white;
      border-radius: 50%;
    }

    #transakCloseIcon:hover,
    #transakCloseIcon:focus {
      color: white;
      background: #${t};
      cursor: pointer;
    }

    #transakIframe{
      width: 100%;
      height: 100%;
      border: none;
      border-radius: 10px;
      background: white;
    }

    @media screen and (max-width: 600px) {
      #transakModal {
        width: 100%;
        height: calc(100% - 24px);
      }

      #transakIframe{
        border-radius: 10px 10px 0 0;
      }
    }
  `}function C(t){let{themeColor:e="1461db",widgetWidth:n="480px",widgetHeight:r="650px"}=t,o=document.createElement("style");return o.innerHTML=I(e,n,r),document.getElementsByTagName("head")[0]?.appendChild(o),o}function k(t,e){let n=C(t),r=document.createElement("div"),o=document.createElement("div"),a=c(d(t));return Object.assign(o,{id:"transakModal",innerHTML:T}),o.appendChild(a),Object.assign(r,{id:"transakRoot",onclick:()=>e()}),r.appendChild(o),document.getElementsByTagName("body")[0].appendChild(r),document.getElementById("transakCloseIcon")?.addEventListener("click",()=>e()),{styleElement:n,rootElement:r,iframeElement:a}}var m=new w.EventEmitter,p=class{#n;#r;#o;#e;#t=!1;static EVENTS=s;constructor(e){if(!e?.widgetUrl)throw new Error("[Transak SDK] => widgetUrl is required");this.#n=e}static on=(e,n)=>{s[e]&&m.on(e,n)};init=()=>{this.#t||(this.#s(),this.#t=!0)};cleanup=()=>{this.#r?.remove(),this.#i(),this.#e?.remove(),this.#t=!1};close=()=>{this.#r?.remove(),this.#o?.remove(),this.#i(),this.#e=void 0,this.#t=!1};#s=()=>{if(window.addEventListener("message",this.#a),this.#n.containerId){let{styleElement:e,iframeElement:n}=u(this.#n);this.#r=e,this.#e=n}else{let{styleElement:e,rootElement:n,iframeElement:r}=k(this.#n,this.#l);this.#r=e,this.#o=n,this.#e=r}};#l=()=>{this.#e?.contentWindow?.postMessage({event_id:"TRANSAK_WIDGET_CLOSE_REQUEST"},"*")};#i=()=>{m.removeAllListeners(),window.removeEventListener("message",this.#a)};#a=e=>{e?.data?.event_id&&this.#t&&m.emit(e.data.event_id,e.data.data)}};export{p as Transak};
